design/move-plugin-repos.md

```diff
@@ -0,0 +1,110 @@
+
+## High-Level Design
+
+Each provider plugin will be an independent project, using the Velero library to implement their specific functionalities.
+
+The way Velero is installed will be changed to accommodate installing these plugins at deploy time, namely the Velero `install` command, as well as the Helm chart.
+
+Each plugin repository will need to have their respective images built and pushed to the same registry as the Velero images.
+
+## Detailed Design
+
+### Projects
+
+Each provider plugin will be an independent GH repository, named: `velero-plugin-aws`, `velero-plugin-azure`, and `velero-plugin-gcp`.
+
+Build of the project will be done the same way as with Velero, using Travis.
+
+Images for all the plugins will be pushed to the same repository as the Velero image, also using Travis.
+
+Releases of new versions can occur as needed and independent of Velero releases. However, a matrix documenting which plugin versions are compatible with each Velero version should be added to the docs.
+
+Documentation for how to install and use the plugins will be augmented in the existing Plugins section of the Velero documentation.
+
+### Installation
+
+As per https://github.com/heptio/velero/issues/1740, we will add a `plugins` flag to the Velero install command which will accept an array of URLs pointing to +1 images of plugins to be installed. The `velero plugin add` command should continue working as is, in specific, it should also allow the installation of any of the new 3 provider plugins. @nrb will provide specifics about how this change will be tackled, as well as what will be documented. Part of the work of adding the `plugins` flag will be removing the logic that adds `velero.io` name spacing to plugins that are added without it. 
+
+The Helm chart that allows the installation of Velero will be modified to accept the array of plugin images with an added `plugins` configuration item..
+
+### Design code changes and considerations
+
+To be decided: what naming convention do we use for name spacing each plugin? Some options:
+
+- prefix it with the same name as the provider: `aws.io/aws`, `azure.io/azure`. `gcp.io/gcp`.
+- leave it with the `velero.io` namespace, since it is maintained by the Velero team.
+
+Install dep
+
+Question: are there any places outside the plugins where we depend on the cloud-provider SDKs? can we eliminate those dependencies too? Answer: still researching. Some data points:
+
+- the `restic` package uses the `aw`. SDK to get the bucket region for the AWS object store (https://github.com/carlisia/velero/blob/32d46871ccbc6b03e415d1e3d4ad9ae2268b977b/pkg/restic/config.go#L41)
+- could not find usage of the cloud provider SDKs anywhere else.
+
+Plugins such as the pod -> pvc -> pv backupitemaction ones make sense to stay in the core repo as they provide some important logic that just happens to be implemented in a plugin.
+
+### Upgrade
```

Comment:
+1 to having some sort of migration command/script. A script would probably be better since it's not tied to the binary and won't need to be removed later